sdk/python/feast/infra/registry/remote.py

```diff
@@ -0,0 +1,369 @@
+from datetime import datetime
+from pathlib import Path
+from typing import List, Optional, Union
+
+import grpc
+from google.protobuf.empty_pb2 import Empty
+from pydantic import StrictStr
+
+from feast.base_feature_view import BaseFeatureView
+from feast.data_source import DataSource
+from feast.entity import Entity
+from feast.feature_service import FeatureService
+from feast.feature_view import FeatureView
+from feast.infra.infra_object import Infra
+from feast.infra.registry.base_registry import BaseRegistry
+from feast.on_demand_feature_view import OnDemandFeatureView
+from feast.project_metadata import ProjectMetadata
+from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto
+from feast.protos.feast.registry import RegistryServer_pb2, RegistryServer_pb2_grpc
+from feast.repo_config import RegistryConfig
+from feast.request_feature_view import RequestFeatureView
+from feast.saved_dataset import SavedDataset, ValidationReference
+from feast.stream_feature_view import StreamFeatureView
+
+
+class RemoteRegistryConfig(RegistryConfig):
+    registry_type: StrictStr = "remote"
+    """ str: Provider name or a class name that implements Registry."""
+
+    path: StrictStr = ""
+    """ str: Path to metadata store.
+    If registry_type is 'remote', then this is a URL for registry server """
+
+
+class RemoteRegistry(BaseRegistry):
+    def __init__(
+        self,
+        registry_config: Union[RegistryConfig, RemoteRegistryConfig],
+        project: str,
+        repo_path: Optional[Path],
+    ):
+        self.channel = grpc.insecure_channel(registry_config.path)
+        self.stub = RegistryServer_pb2_grpc.RegistryServerStub(self.channel)
+
+    def apply_entity(self, entity: Entity, project: str, commit: bool = True):
+        raise Exception("Unimplemented: remote registry is read-only")
```

Comment:
consider adding new exception class in `errors.py`?

Reply:
fixed it